boards/arm/arduino_portenta_h7/arduino_portenta_h7_m7_defconfig

```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_STM32H7X=y
+CONFIG_SOC_STM32H747XX=y
+
+# Board config should be specified since there are 2 possible targets
+CONFIG_BOARD_ARDUINO_PORTENTA_H7_M7=y
+
+# Enable the internal SMPS regulator
+CONFIG_POWER_SUPPLY_DIRECT_SMPS=y
+
+# Enable GPIO
+CONFIG_GPIO=y
+
+# Enable clocks
+CONFIG_CLOCK_CONTROL=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
+
+# Enable HW stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# enable pin controller
+CONFIG_PINCTRL=y
+
+# Use zephyr,code-partition as flash offset
+CONFIG_USE_DT_CODE_PARTITION=y
```

Comment:
Uncommon, is that made on purpose ?

Reply:
From my understanding of [this doc](https://docs.zephyrproject.org/latest/develop/flash_debug/host-tools.html#sam-boot-assistant-sam-ba) (*Without ROM bootloader" tab), this board has a bootloader at the start of the flash so this config allows to set the partition table in the device tree.